### DISCUSSION TITLE
Auto Squaring Question

### DISCUSSION THREAD

**Discussion by rvalotta** [terjeio/grblHAL#316] [Q&A]
Is there any way for me to confirm that Auto Squaring is compiled in?    When I home my Y axis it goes back about 2/3 of the way,, stops and I get an error 15.   Even when I get it all the way to the back with a intentional skew and homes the Y axis but stops there and does nothing with Y2 at that point. 

STM32f411

**Comment by terjeio** [comment]
You have some options for that:

* Only one limit input needs to be triggered to show the axis letter in the status report.

* The new `$pins` report lists the pins for the second motor.

* The `$17x` setting is available - `$171` for the Y-axis, you can use `$help y-axis` to check.

**Reply by rvalotta** [reply]
okay so $pins shows me the following

[PIN:PB6,Reset]
[PIN:PB7,Feed hold]
[PIN:PB8,Cycle start]
[PIN:PB15,Probe]
[PIN:PB12,X limit min]
[PIN:PB13,Y limit min]
[PIN:PB10,Y limit min 2]
[PIN:PB14,Z limit min]
[PIN:PA0,X step]
[PIN:PA2,Y step]
[PIN:PA4,Z step]
[PIN:PA9,Y step]
[PIN:PA1,X dir]
[PIN:PA3,Y dir]
[PIN:PA5,Z dir]
[PIN:PA10,Y dir]
[PIN:PB0,Steppers enable]
[PIN:PB1,Spindle on]
[PIN:PB2,Spindle direction]
[PIN:PC15,Mist]
[PIN:PC14,Flood]

and $help y-axis shows 

$101: Y-axis travel resolution in step/mm
$111: Y-axis maximum rate in mm/min
$121: Y-axis acceleration in mm/sec^2
$131: Y-axis maximum travel in mm
$171: Y-axis dual axis offset in mm, range: -2 - 2

So it looks like it is enabled...   My Y limit hits and I don't see any continued movement on y2.   Makes me believe that I missed something in my code changes to make this work properly.  Any idea's where I should look?

**Reply by rvalotta** [reply]
Also on a side note in iosender under the section named Dual Axis, the help for Dual axis length fail max reads Dual axis length fail minimum distance.  I think that should be maximum distance. 😄

**Reply by terjeio** [reply]
>  I think that should be maximum distance.

Thats a typo, I'll fix it in the next release.

I have decided to do the job of moving definition of ganged/auto squared axes to my_machine.h - and I have spend all day to get the basic stuff in place. With luck your board file should now be ok with that, and I want you to test it so I have attached the project as it is now. Please note that it will be a while before I can commit to github as currently only one extra axis is supported - I'll see if I can do some preprocessor trickery to make it simple for more... The main preprocessor stuff is in grbl/motors.h if you want to take a look.

[GRBL Driver STM32F401.zip](https://github.com/terjeio/grblHAL/files/6656574/GRBL.Driver.STM32F401.zip)

**Reply by phil-barrett** [reply]
With my new Avid machine, I will be doing a blog on autosquaring so this is a timely change.  Hopefully, it will apply to all grblHAL ports.  Will probably be bugging you for comments on it before I go public.

**Reply by terjeio** [reply]
> Hopefully, it will apply to all grblHAL ports.

It will, and other changes will make it easier to reassign unused stepper pins to aux I/O (at compile time as a first step).

**Reply by rvalotta** [reply]
Thanks I've put it on a spare STM32 and will pop it in late tonight and get back to you.

**Reply by rvalotta** [reply]
@terjeio Thanks that seems to have fixed the auto square issue... I need to review my Z limit config as that is not working as expected but to be fair that limit switch was a B17ch to get installed so i may of fed it up.

@phil-barrett Is there a way I can get intouch with you to discuss something that is not directly realated to grblhal.

**Reply by phil-barrett** [reply]
You can contact me through [the tindie store](https://www.tindie.com/products/philba/grblhal-breakout-board-unkit-for-teensy-41/)
